geoapps/iso_surfaces/driver.py

```diff
@@ -158,9 +160,13 @@ def iso_surface(
             )
             values = values[0].reshape(x.shape)
 
+        lower, upper = np.nanmin(values), np.nanmax(values)
         surfaces = []
-        for level in levels:
+        print("Running marching cubes on levels.")
+        for level in tqdm(levels):
             try:
+                if level < lower or level > upper:
+                    continue
```

Comment:
Maybe log that the levels are being skipped?

Reply:
Good call. Done

Reply:
That should do it!